Src/GoogleApis.Auth/OAuth2/ClientCredentialParameters.cs

```diff
@@ -0,0 +1,105 @@
+        ServiceAccountCredential = 2
+    }
+
+    /// <summary>Container for the union of parameters/properties from all credential types.</summary>
+    public sealed class ClientCredentialParameters
+    {
+        /// <summary>Indicates whether the credentials is of type ServiceAccountCredential or UserCredential.</summary>
+        [Newtonsoft.Json.JsonProperty("type")]
+        private string Type { get; set; }
+
+        /// <summary>Client Email associated with ServiceAccountCredential obtained from <a href="https://console.developers.google.com">Google Developers Console</a></summary>
+        [Newtonsoft.Json.JsonProperty("client_email")]
+        public string ClientEmail { get; set; }
+
+        /// <summary>Cliend Id associated with UserCredential created by <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("client_id")]
+        private string ClientId { get; set; }
+
+        /// <summary>Client Secret associated with UserCredential created by <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("client_secret")]
+        private string ClientSecret { get; set; }
+
+        /// <summary>Private Key associated with ServiceAccountCredential obtained from <a href="https://console.developers.google.com">Google Developers Console</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("private_key")]
+        public string Pkcs8PrivateKey { get; set; }
+
+        /// <summary>Refresh Token associated with UserCredential created by <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("refresh_token")]
+        public string RefreshToken { get; set; }
+
+        /// <summary>Gets the client secrets which contains the client identifier and client secret. </summary>
+        public ClientCredentialType CredentialType
+        {
+            get
+            {
+                Utilities.ThrowIfNullOrEmpty(Type, "Type");
+
+                switch (Type)
+                {
+                    case "authorized_user":
+                        return ClientCredentialType.UserCredential;
+
+                    case "service_account":
+                        return ClientCredentialType.ServiceAccountCredential;
+
+                    default:
+                        throw new InvalidOperationException(String.Format("Error parsing stream contents. Unrecognized Credential Type {0}.", Type));
+                }
+            }
+        }
+
+        /// <summary>Gets the ClientSecrets portion of the credential parameters.</summary>
+        public ClientSecrets ClientSecrets
+        {
+            get
+            {
+                Utilities.ThrowIfNullOrEmpty(ClientId, "ClientId");
```

Comment:
I don't see this pattern being used in other places in the existing code where this method is used. Any reason why you would want it to be used here?